### ISSUE TITLE
clang-format gives wonky indentation

### ISSUE THREAD

**Issue by epixinvites** [llvm/llvm-project#54354]
`clang-format version 13.0.1`
clang-format options -> https://termbin.com/n7gx

So, my problem here is that whenever I format a piece of code, similar to the below
```cpp
std::vector<std::tuple<int, int, int>> input {	// Node a, node b, weightage
	{1, 2, 5},
	{1, 3, 3},
	{1, 4, 7},
	{2, 4, 3},
	{2, 5, 2},
	{3, 4, 1},
	{4, 5, 2}
};
```
clang-format gives me some wretched output like this
```cpp
	std::vector<std::tuple<int, int, int>> input {
  // Node a, node b, weightage
	{1, 2, 5},
    {1, 3, 3},
    {1, 4, 7},
    {2, 4, 3},
    {2, 5, 2},
    {3, 4, 1},
    {4, 5, 2}
    };
```
A horrendous mixture of tabs and spaces and the comments behind are somehow thrown to the next line with 0 indentation.

Is it something to do with my wrongly configured clang-format? If so, please help me point out what I have to change in order to achieve the formatting on the first code block.

**Comment by HazardyKnusperkeks** [comment]
You should give your .clang-format file.

**Comment by epixinvites** [comment]
@HazardyKnusperkeks I literally linked it in the second line of my issue?

**Comment by HazardyKnusperkeks** [comment]
It was late, I did not look for links, but for text in the issue.

**Comment by epixinvites** [comment]
> It was late, I did not look for links, but for text in the issue.

Ah, it's alright. I've just been scrolling through the clang-format options' documentation for like hours and really fed up.

**Comment by mydeveloperday** [comment]
Standard LLVM format would give this

```c
std::vector<std::tuple<int, int, int>> input{// Node a, node b, weightage
                                             {1, 2, 5}, {1, 3, 3}, {1, 4, 7},
                                             {2, 4, 3}, {2, 5, 2}, {3, 4, 1},
                                             {4, 5, 2}};
```

I get this with your clang-format file

```c
std::vector<std::tuple<int, int, int>> input {
  // Node a, node b, weightage
{1, 2, 5},
{1, 3, 3},
{1, 4, 7},
{2, 4, 3},
{2, 5, 2},
{3, 4, 1},
{4, 5, 2}
};
```

**Comment by epixinvites** [comment]
Yeah... and I've looked through the documentation, I can't find any configuration that allows me to change the formatting behaviour.

**Comment by mydeveloperday** [comment]
Are you expecting to see it untouched from your original? If I put this into your clang-format I get this

AlignArrayOfStructures: Left
ColumnLimit: 200

```
std::vector<std::tuple<int, int, int>> input{
  // Node a, node b, weightage
    {1, 2, 5},
    {1, 3, 3},
    {1, 4, 7},
    {2, 4, 3},
    {2, 5, 2},
    {3, 4, 1},
    {4, 5, 2}
};
```

**Comment by mydeveloperday** [comment]
but if I add back in BasedOnGNU

```c
std::vector<std::tuple<int, int, int> > input{
  // Node a, node b, weightage
  {1,  2, 5},
  { 1, 3, 3},
  { 1, 4, 7},
  { 2, 4, 3},
  { 2, 5, 2},
  { 3, 4, 1},
  { 4, 5, 2}
};
```

Something else go on!

**Comment by epixinvites** [comment]
> but if I add back in BasedOnGNU
> 
> ```c
> std::vector<std::tuple<int, int, int> > input{
>   // Node a, node b, weightage
>   {1,  2, 5},
>   { 1, 3, 3},
>   { 1, 4, 7},
>   { 2, 4, 3},
>   { 2, 5, 2},
>   { 3, 4, 1},
>   { 4, 5, 2}
> };
> ```
> 
> Something else go on!

Aha, but the thing is if you look carefully at the second line in the clang-format I've commented the `BasedOnStyle: GNU` out.

**Comment by mydeveloperday** [comment]
I know but that would turn on a load of options that we cannot detect

**Comment by mydeveloperday** [comment]
For example

```yaml
AlignArrayOfStructures: Left
ColumnLimit: 200
Cpp11BracedListStyle: true
SpaceBeforeCpp11BracedList: true
UseTab: AlignWithSpaces
```

is pretty good...

```c
std::vector<std::tuple<int, int, int>> input {
  // Node a, node b, weightage
    {1, 2, 5},
    {1, 3, 3},
    {1, 4, 7},
    {2, 4, 3},
    {2, 5, 2},
    {3, 4, 1},
    {4, 5, 2}
};
```

**Comment by mydeveloperday** [comment]
Some between these 5 options and your full .clang-format file is the option that is interfering...if you want to help me out, add them back in one by one, until it goes "wonky"... then start removing the ones that make no difference, and we'll likely end up with a couple of option reproducer.

**Comment by mydeveloperday** [comment]
be aware I'm using clang-format 15 and we made a change to try and fix AlignArrayOfStructures to prevent the crashes, so you either want to try that or be careful.

**Comment by mydeveloperday** [comment]
Also we need to be clear your original has tabs in right...

![image](https://user-images.githubusercontent.com/16046346/159118022-afe007a1-922c-4b14-902f-92f3cd202382.png)

**Comment by epixinvites** [comment]
> Some between these 5 options and your full .clang-format file is the option that is interfering...if you want to help me out, add them back in one by one, until it goes "wonky"... then start removing the ones that make no difference, and we'll likely end up with a couple of option reproducer.

Alright, thank you so much, at least now I have something to work and test on.

**Comment by epixinvites** [comment]
After some testing, setting `ContinuationIndentWidth: 4` seems to fix the wonky indentation, but I seem to not be able to align the comments.
Here's the new `.clang-format` -> https://termbin.com/fdn2

This is the result:
```cpp
    std::vector<std::pair<int, int>> graphInput {
  // a comment
        {1, 2},
        {1, 3},
        {1, 4},
        {1, 5},
        {2, 6},
        {4, 7},
        {4, 8},
        {4, 9}
    };
```
I've fixed it to indent by 4 spaces, but I can't fix the comments.

**Comment by Hedede** [comment]
Your termbin links are dead.

**Comment by epixinvites** [comment]
> @epixinvites Your termbin links are dead.

Yeah it expires after a month and I'm pretty busy these days, gotta deal with the wonky indentation until I find some a block of time to do some debugging.
